### PR TITLE
Reject KREFLECTION.isConcrete hook

### LINKED ISSUE #1367: Reject KREFLECTION.isConcrete hook

### PR DESCRIPTION
**Pull request by andreiburdusa**
---
Fixes #1367 
###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/src/Kore/Builtin/Builtin.hs [hunk 1]
```diff
@@ -131,6 +133,39 @@ import qualified Kore.Unification.Unify as Monad.Unify
...
+    [ ( "KREFLECTION.isConcrete"
+      , verifySymbolKreflection assertSort [assertSort, assertSort]
+      )
+    ]
+
+verifySymbolKreflection
+    :: SortVerifier  -- ^ Builtin result sort
+    -> [SortVerifier]  -- ^ Builtin argument sorts
+    -> SymbolVerifier
+verifySymbolKreflection _ _ =
```
Comment (ttuegel) [inline kore/src/Kore/Builtin/Builtin.hs]: I suggest renaming this to `rejectSymbolIsConcrete`. It should not take any arguments, and it should not be necessary to define `kreflectionSort` or `assertSort`.

File: kore/src/Kore/Builtin/Builtin.hs [hunk 2]
```diff
@@ -17,6 +17,7 @@ builtin modules.
 module Kore.Builtin.Builtin
     ( Function
     , FunctionImplementation
+    , verifiers
```
Comment (ttuegel) [inline kore/src/Kore/Builtin/Builtin.hs]: This is not an appropriate place to define these verifiers. Please make a new module.

File: kore/src/Kore/Builtin/Endianness.hs [hunk 3]
```diff
@@ -27,7 +27,9 @@ import Data.String
     )
 
 import qualified Kore.Attribute.Symbol as Attribute.Symbol
-import Kore.Builtin.Builtin
+import Kore.Builtin.Builtin hiding
```
Comment (virgil-serbanuta) [inline kore/src/Kore/Builtin/Endianness.hs]: Not sure how would that look, but, if the list would not bee too long, why not import only what's needed?

The same in Signedness.hs

**Comment by andreiburdusa** [comment]
I still don't know how to convert the error caused by the sort rejection into a passing test.
I'll read about makefiles on Monday and maybe I'll figure it out

**Comment by andreiburdusa** [comment]
The K integration tests are failing, but I don't think it's because the one I added

File: kore/src/Kore/Builtin/Kreflection.hs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+symbolVerifiers :: Builtin.SymbolVerifiers
+symbolVerifiers =
+    HashMap.fromList
+    [ ( "KREFLECTION.isConcrete" , rejectSymbolIsConcrete)
+    ]
+
+rejectSymbolIsConcrete :: Builtin.SymbolVerifier
+rejectSymbolIsConcrete =
+    Builtin.SymbolVerifier $ \_ _ -> Left $ Kore.Error.Error
```
Comment (ttuegel) [inline kore/src/Kore/Builtin/Kreflection.hs]:
```suggestion
    Builtin.SymbolVerifier $ \_ _ -> Kore.Error.koreFail "found KREFLECTION.isConcrete hook"
```

File: kore/src/Kore/Builtin/Kreflection.hs [hunk 5]
```diff
@@ -0,0 +1,33 @@
+{- |
+Module      : Kore.Builtin.Kreflection
+Copyright   : (c) Runtime Verification, 2019
```
Comment (virgil-serbanuta) [inline kore/src/Kore/Builtin/Kreflection.hs]: 2020?

File: test/kreflection/test-kreflection.sh [hunk 6]
```diff
@@ -0,0 +1,9 @@
+TOP=$(git rev-parse --show-toplevel)
+KRUN=$TOP/.build/k/bin/krun
+KORE_EXEC=$TOP/.build/kore/bin/kore-exec
+KRUN_OPTS="--haskell-backend-command \"$KORE_EXEC --disable-log-timestamps\""
+COMMAND="$KRUN $KRUN_OPTS -d . 1.kreflection 2>test-kreflection.sh.out || true"
+KOMPILE="$TOP/.build/k/bin/kompile kreflection.k --backend haskell"
+eval $KOMPILE
```
Comment (ttuegel) [inline test/kreflection/test-kreflection.sh]: You should not run `kompile` here, but let `make` handle the compilation by including a dependency on `KOMPILED`.

File: test/kreflection/Makefile [hunk 7]
```diff
@@ -0,0 +1,3 @@
+include $(CURDIR)/../include.mk
+
+test-%.sh.out: $(TEST_DIR)/test-%-*
```
Comment (ttuegel) [inline test/kreflection/Makefile]:
```suggestion
test-%.sh.out: $(TEST_DIR)/test-%-* $(KOMPILED)
```

File: test/kreflection/test-kreflection.sh [hunk 8]
```diff
@@ -0,0 +1,9 @@
+TOP=$(git rev-parse --show-toplevel)
```
Comment (ttuegel) [inline test/kreflection/test-kreflection.sh]: You must not set `TOP`, `KRUN`, `KORE_EXEC`, or `KRUN_OPTS` here. These must be set by `make`. If a variable you need is not set by `make`, add an `export` declaration to the `Makefile`, such as: `export KRUN`.

**Pull request merged**